action.yaml

```diff
@@ -362,23 +362,26 @@ runs:
           cd ${{ inputs.root-folder }}
         fi
 
-        branch=$(echo "${GITHUB_REF#refs/heads/}")
-        echo "Branch pushed to: $branch"
-        git fetch origin $branch
-
         DAGS_ONLY_DEPLOY=false
-        SKIP_IMAGE_OR_DAGS_DEPLOY=true
+        SKIP_IMAGE_OR_DAGS_DEPLOY=false
+        files=()
 
-        # case when the triggered event is a manual workflow dispatch or a new branch or tag creation, we would need to deploy the image because we cannot determine that it does not need to be deployed
         GITHUB_EVENT_BEFORE=${{ github.event.before }}
         GITHUB_EVENT_AFTER=${{ github.event.after }}
+        # case when the triggered event is a manual workflow dispatch or a new branch or tag creation, we would need to deploy the image because we cannot determine that it does not need to be deployed
         if [[ "$GITHUB_EVENT_BEFORE" == "0000000000000000000000000000000000000000" ||  -z $GITHUB_EVENT_BEFORE && -z $GITHUB_EVENT_AFTER ]]; then
-            DAGS_ONLY_DEPLOY=false
-            SKIP_IMAGE_OR_DAGS_DEPLOY=false
-            files=()
+          echo "Manual workflow dispatch or a new branch or tag creation, hence missing github event before and/or after commit hash"
         else
-          files=$(git diff --name-only $GITHUB_EVENT_BEFORE $GITHUB_EVENT_AFTER)
-          echo "files changed: $files"
+          echo "event that triggered the workflow: $GITHUB_REF"
+          branch=$(echo "${GITHUB_REF#refs/heads/}")
+          git fetch origin $branch
+          if ! git cat-file -e "${{ github.event.before }}" 2>/dev/null; then
```

Comment:
in case of force push, we would get a commit hash, which won't exist in the git history due to force push